proxy-node-gateway/src/main/java/uk/gov/ida/notification/saml/validation/EidasAuthnRequestValidator.java

```diff
@@ -46,6 +84,18 @@ public void validate(AuthnRequest request) {
         spTypeValidator.validate(spTypeElement);
         loaValidator.validate(request.getRequestedAuthnContext());
         requestedAttributesValidator.validate(requestedAttributesElement);
+        assertionConsumerServiceValidator.validate(request);
+        comparisonValidator.validate(request.getRequestedAuthnContext());
+
+        if (!duplicateAuthnRequestValidator.valid(request.getID())) {
```

Comment:
the `duplicateAuthnRequestValidator` keep track of request ID's in an in memory map. If we have more than one instance running this validation becomes flakey. Also, if an instance is restarted we lose the current list. This map will just continues to grow and consume memory. Same issues that https://govukverify.atlassian.net/browse/EID-1117 is looking to resolve.

Also, why 5 minutes as the time limit for checking duplication? Shouldn't the ID's be unique - why have a time limit?

Reply:
Yes this is the same problem. We decided to "fix it later" rather than sort it now. Sorting this is in scope for EID-1117.

AuthnRequests come from the service – I think the duplication window is there to allow the user's browser to resubmit if there are network errors, they press Back, etc. 5 minutes is a fairly aribtrary value that we need to pick a more intelligent value for in production.